### PR TITLE
remove deprecated prog_load API.

### LINKED ISSUE #1146: Change socket_tests.cpp to use bpf_object__open_file etc.  instead of  deprecated API.
Can you please change it to use bpf_object__open_file etc.  instead of using a deprecated API?
And then maybe we can remove the deprecated API implementation altogether?

_Originally posted by @shankarseal in https://github.com/microsoft/ebpf-for-windows/pull/1130#discussion_r879629317_

### PR DESCRIPTION
**Pull request by shankarseal**
## Description

Fixes #1146 

## Testing

No new tests needed.

## Documentation

Doxygen will automatically update API docs.

### CODE REVIEW

**Review by Alan-Jowett — APPROVED** [review]

File: ebpfapi/Source.def [hunk 1]
```diff
@@ -67,7 +67,6 @@ EXPORTS
     bpf_prog_detach2
     bpf_prog_get_fd_by_id
     bpf_prog_load
-    bpf_prog_load_deprecated
```
Comment (dthaler) [inline ebpfapi/Source.def:70]: revert this line.  Need to leave this in or dependent code will break

File: tests/unit/libbpf_test.cpp [hunk 2]
```diff
@@ -315,12 +315,10 @@ TEST_CASE("libbpf program", "[libbpf]")
 {
     _test_helper_libbpf test_helper;
 
-    struct bpf_object* object;
-    int program_fd;
-    int result = bpf_prog_load_deprecated("droppacket.o", BPF_PROG_TYPE_XDP, &object, &program_fd);
-    REQUIRE(result == 0);
+    struct bpf_object* object = bpf_object__open("droppacket.o");
```
Comment (dthaler) [inline tests/unit/libbpf_test.cpp:318]: These changes are fine but we also need to add a test case with the tags "[libbpf][deprecated]" that explicitly still uses `bpf_prog_load_deprecated`

**Comment by dthaler** [comment]
This PR also addresses one of the checkboxes of #1295

File: tests/unit/libbpf_test.cpp (hunk 2 above)
Comment (shankarseal) [reply tests/unit/libbpf_test.cpp:318]: I get `'bpf_prog_load_deprecated': Use bpf_object__open() and bpf_object__load() instead. [F:\ebpf-for-windows\my\tests\unit\test.vcxproj]` due to the way the function is annotated.

Comment (dthaler) [reply tests/unit/libbpf_test.cpp:318]: Correct, you need to add the pragma like line 296 does for the same reason.  Every test in this file with the `[deprecated]` tag has a line like that.

**Review by dthaler — APPROVED** [review]

**Comment by dthaler** [comment]
@shankarseal please resolve conflicts

**Review by Alan-Jowett — DISMISSED** [review]

**Review by dthaler — APPROVED** [review]

**Review by Alan-Jowett — APPROVED** [review]

**Pull request merged**